[CDAP-18627] Switch DefaultPreviewStore to use new LevelDBTableCore API

wyzhang opened the pull request:
Why:
DefaultPreviewStore doesn't need to read/write/delete to levelDB at different versions.
So switch to use new LevelDBTableCore API that operate at max version (i.e. always
overwrite old value). This also provide some performance gain by getRow()->get()
as getRow() internally perform iterator().next while get() is a point lookup
for a specific version.

tivv commented on cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/preview/DefaultPreviewStore.java:
```diff
@@ -216,16 +216,17 @@ public PreviewStatus getPreviewStatus(ApplicationId applicationId) {
-      return gson.fromJson(Bytes.toString(row.get(STATUS)), PreviewStatus.class);
+
+    if (status != null) {
+      return gson.fromJson(Bytes.toString(status), PreviewStatus.class);
```
nit: use `new InputStreamReader(new ByteArrayInputStream(status), UTF-8)` instead of  making a copy

tivv replied:
Similarly reverse conversions are pretty inefficient :(

wyzhang committed:
[CDAP-18627] Switch DefaultPreviewStore to use new LevelDBTableCore API
             Also fix a test breakage for DefaultPreviewStoreTest

Why:
DefaultPreviewStore doesn't need to read/write/delete to levelDB at different versions.
So switch to use new LevelDBTableCore API that operate at max version (i.e. always
overwrite old value). This also provide some performance gain by getRow()->get()
as getRow() internally perform iterator().next while get() is a point lookup
for a specific version.
